styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesDockerRunner.java

```
@@ -111,106 +125,139 @@
-        .build());
-    env.add(new EnvVarBuilder()
-        .withName(EXECUTION_ID)
-        .withValue(podName)
-        .build());
-    env.add(new EnvVarBuilder()
-        .withName(TERMINATION_LOG)
-        .withValue("/dev/termination-log")
-        .build());
-    env.add(new EnvVarBuilder()
-        .withName(TRIGGER_ID)
-        .withValue(runSpec.trigger().map(TriggerUtil::triggerId).orElse(null))
-        .build());
-    env.add(new EnvVarBuilder()
-        .withName(TRIGGER_TYPE)
-        .withValue(runSpec.trigger().map(TriggerUtil::triggerType).orElse(null))
-        .build());
-
-    PodBuilder podBuilder = new PodBuilder()
+    final PodBuilder podBuilder = new PodBuilder()
         .withNewMetadata()
         .withName(podName)
         .addToAnnotations(STYX_WORKFLOW_INSTANCE_ANNOTATION, workflowInstance.toKey())
         .addToAnnotations(DOCKER_TERMINATION_LOGGING_ANNOTATION, String.valueOf(runSpec.terminationLogging()))
         .endMetadata();
-    PodFluent.SpecNested<PodBuilder> spec = podBuilder.withNewSpec()
-        .withRestartPolicy("Never");
-    PodSpecFluent.ContainersNested<PodFluent.SpecNested<PodBuilder>> container = spec
+    final PodFluent.SpecNested<PodBuilder> spec = podBuilder.withNewSpec().withRestartPolicy("Never");
+    final PodSpecFluent.ContainersNested<PodFluent.SpecNested<PodBuilder>> container = spec
         .addNewContainer()
             .withName(STYX_RUN)
             .withImage(imageWithTag)
             .withArgs(runSpec.args())
             .withEnv(env);
 
+    if (runSpec.serviceAccount().isPresent()) {
+      spec.addNewVolume()
+          .withName(STYX_WORKFLOW_SA_SECRET_NAME)
+          .withNewSecret()
+          .withSecretName(buildSecretName(runSpec.serviceAccount().get()))
+          .endSecret()
+          .endVolume();
+      container
+          .addToVolumeMounts(new VolumeMountBuilder()
+              .withMountPath(STYX_WORKFLOW_SA_SECRET_MOUNT_PATH)
```

What happens in the unlikely case that the mounting path is the same of the one for the old-fashioned secret and both are mounted? If that not allowed, should we throw an exception with a custom message as we do, for example, in case of failure in creating the keys?